plugin/pkg/admission/priority/apis/priority/types.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package priority
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Configuration provides configuration for the priority admission controller.
+type Configuration struct {
+	metav1.TypeMeta
+
+	// List of namespaces allowed to have system critical priority classes
+	Namespaces []string
```

Comment:
While this may work fine, I thought Derek wanted to have something like:

```go
type priorityClassToNamespace struct {
   priorityClassPrefixes  []string
   namespacePrefixes    []string
}

type Configuration struct {
    ...
    allowedPriorityClasses []priorityClassToNamespace
}
```

Then we can have something like:

allowedPriorityClasses: {{system-},{kube-system}} to allow all system priority classes in kube-system or
allowedPriorityClasses: {{system-node-critical, system-cluster-critical},{kube-system, openshift-}} to allow the two system priority classes in kube-system and any namespace prefixed with openshift-.

This is probably less secure, but more flexible.

Reply:
Yeah if we `really want/need`, I can do this. I just wanted to keep configuration simple. Complex/more flexible configurations are always error/confusion prone. 

Lets see feedback from others, if there is an agreement, I would be happy to update it.

Reply:
I tend to agree with @bsalamat, there’s a use case where we want to apply this for prefixed namespaces (these namespaces are created dynamically)

Reply:
Assuming we go with approach, I'd prefer grouping based on per priority class. 

For example, in the above, lets say, p1,p2 are allowed in n1,n2 but p1 is allowed in n3 too. This means, it would require configuring: {{p1,p2},{n1,n2}},{{p1},{n3}}. That means for any priority class, we will have to go through the whole mapping. Also at the time of configuration, it might be easy to miss if a particular class has already been configured for a namespace already or not, or might lead to duplication. 

Due to above reasons, I'd suggest, configuration based on per priority class, and that means, the above example would look something like: {{p1}, {n1,n2,n3}}, {{p2},{n1,n2}}. This way we can directly map a particular class to set of allowed namespaces. Also, it would be less error prone at the time of configuration.

The other thing we need to think, if we want to allow, only full names of priority classes, or we would like to have more general/flexible grouping based on priority class prefix. Though, I would want to avoid it, but In this case, again, we will have to check if a particular priority class has been configured by its full name or is part of some prefix, and there is no conflict between them. Or set of allowed namespaces based on full name would always override set of allowed namespaces based on its prefix.